src/BuiltInTools/dotnet-watch/Internal/HotReloadFileSetWatcher.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.Watcher.Internal
+{
+    public class HotReloadFileSetWatcher : IDisposable
```

Comment:
@BrennanConroy the most significant change is to this type. Could I have you look at this?